### DISCUSSION TITLE
Continue recording during brief (taxi) stops

### DISCUSSION THREAD

**Discussion by sampie777** [SAHorowitz/MSFS2020-PilotPathRecorder#16] [Ideas]
Hi there,

Is it possible to add some sort of adjustable cooldown period before ending the flight recording? For example, when I need to halt while taxiing (e.g. before a runway), the recorder will stop the recording and start  a new one when I resume taxi. This creates multiple recordings for a single flight, which looks quite messy. 

My settings are record interval = 1 sec and minimum speed = 1 knots. When I set minimum speed to 0 knots, it already starts recording while in the main menu which adds incorrect coordinates to the recording.

**Comment by SAHorowitz** [comment]
How about only stop logging when the parking brake is applied?  This would be instead of a cool-down period.

**Reply by sampie777** [reply]
Hm, not sure as one could set the parking brake for doing pre-takeoff inspections haha. Parking brake + engine off + battery off might work? But how would one handle just randomly exiting while mid-flight, because then no parking brake or whatsoever will be set. I don't know if FS2020 has a parameter which indicates if it's in the main menu or not. I know the old sims didn't have such a thing, except for the 'simulation paused' flag. I see there's a SimStart and SimStop parameter in MSFS2020_SimConnectIntergration.cs, but I don't know if that's what I think it might be?

**Comment by SAHorowitz** [comment]
Maybe I am still misunderstanding. I think you asked for auto start logging like it works today. Once you go above a certain speed start logging. However, what I think you asked for was don't stop logging when below the speed because you like to record the taxing, etc. So I think if the app adds an option to only stop recording when parking brake is on would work. Would it not?

In terms of stopping recording when user goes to menu, that is not possible currently but the app does have a pause and resume logging for that purpose.

Let me know if what I propose would work for you.

**Reply by sampie777** [reply]
I'm talking about "auto stop logging". 

My problem is: when I need to halt during taxi, the current log is stopped. When I continue taxi, a new log is started. 

What I try to suggest: merge these two logs into one log. 

Your solution with the parking brake will be a first fix. I just think that checking the parking brake is not enough, as some people might use the parking brake during the taxi halt. Thus still creating separate logs. But for now, as you want to release the new version, go ahead with implementing the parking brake solution. I will be happy with that. We can talk again after the release about a more solid solution :)

**Comment by SAHorowitz** [comment]
Thought of another idea. What if I give an option to use engine 1 to 4 RPM as 0 as an indicator to stop logging? It would be an "engine off to stop auto-logging option." This would be instead of the parking brake. This would also be a suboption to the auto-logging just based on speed. So for people who just want speed as an auto-logging start/stop they can leave everything as is. For people that want to continue to log even after slowing down below threshold as long as one of the engines has a greater than zero RPM it will keep recording. Thoughts?

**Reply by sampie777** [reply]
I love it!

**Comment by SAHorowitz** [comment]
1.4.0 has been released with this functionality in it.
